torch/distributed/_shard/sharding_spec/chunk_sharding_spec_ops/_common.py

```diff
@@ -406,24 +248,22 @@ def _handle_max_norm_col_wise(
         local_shard: col-wise shared local weight used for lookup.
         input: tensor to be applied op to.
         world_size: number of ranks.
+        gathered_inputs: list of inputs from all ranks.
         pg: process group.
 
     Return:
         local_shard_norm_renormed: local_shard re-normed to max_norm if the norm is larger
             than it.
-        gathered_inputs: list of inputs from all ranks.
+
     """
     norm_type = norm_type if norm_type is not None else 2.0
-    # allgather the inputs first.
-    gathered_inputs = [torch.zeros_like(input) for _ in range(world_size)]
-    dist.all_gather(gathered_inputs, input, group=pg)
     unique_inp = torch.unique(torch.cat(gathered_inputs))
```

Comment:
Let me think about this..

Reply:
After a second thought, we probably need to keep using all_gather for col-wise sharding because we need to make it a list and do all the rearrangement. However, when we migrating to DT, this should not be a problem at all.